sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/ColumnPruningSupport.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.sources.v2.reader;
+
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * A mix in interface for `DataSourceV2Reader`. Users can implement this interface to only read
+ * required columns/nested fields during scan.
+ */
+public interface ColumnPruningSupport {
+  /**
+   * Returns true if the implementation can apple this column pruning optimization, so that we can
+   * reduce the data size to be read at the very beginning.
+   */
+  boolean pruneColumns(StructType requiredSchema);
```

Comment:
Why allow the implementation to reject the required schema? If the data source supports column pruning, then it should throw an exception if the columns passed by this method are invalid (with more context). Otherwise, it should do the pruning. The only case where I think this might be valid is if there are some columns that can't be pruned, in which case it should return the actual schema that will be produced instead of rejecting by returning false. But then, the actual read schema is accessible from `DataSourceV2Reader#readSchema`.

Reply:
Actually I don't have a use case for rejecting the required schema, just wanna follow other push down interface to return a boolean. I'm fine to return `void` here.